http/httputils.c

```diff
@@ -0,0 +1,564 @@
+ */
+static gvm_http_multi_t *
+gvm_http_multi_new (void)
+{
+  return (gvm_http_multi_t *) g_malloc0 (sizeof (struct gvm_http_multi));
+}
+
+/**
+ * @brief Callback function to store the response.
+ * This function is used with libcurl to dynamically store response data.
+ */
+static size_t
+curlutils_response_callback (void *ptr, size_t size, size_t nmemb, void *userdata)
+{
+  gvm_http_response_stream_t stream = userdata;
+  size_t new_len = stream->length + size * nmemb;
+  gchar *temp_ptr = g_realloc (stream->data, new_len + 1);
+
+  if (!temp_ptr)
+    return 0;
+
+  stream->data = temp_ptr;
+  memcpy (stream->data + stream->length, ptr, size * nmemb);
+  stream->data [new_len] = '\0';
+  stream->length = new_len;
+
+  return size * nmemb;
+}
+
+/**
+ * @brief Allocates and initializes a gvm_http_t structure with a given CURL handle.
+ *
+ * @param curl_handler  A valid libcurl easy handle to be wrapped. If NULL, the function returns NULL.
+ *
+ * @return A pointer to an initialized `gvm_http_t` structure on success,
+ *         or NULL if the input handle is invalid.
+ */
+gvm_http_t *
+gvm_http_new (CURL *curl_handler)
+{
+  if (!curl_handler)
+    return NULL;
+
+  gvm_http_t *http = g_malloc0 (sizeof (gvm_http_t));
+  http->handler = curl_handler;
+  return http;
+}
+
+/**
+ * @brief Frees a gvm_http_t object and its associated CURL handle.
+ *
+ * @param http Pointer to the gvm_http_t structure to free. Safe to pass NULL.
+ */
+void
+gvm_http_free (gvm_http_t *http)
+{
+  if (!http) return;
+  if (http->handler)
+    curl_easy_cleanup (http->handler);
+  g_free (http);
+}
+
+/**
+ * @brief Initializes and configures a gvm_http_t object for an HTTP(S) request.
+ *
+ * This function creates and configures a gvm_http_t structure, encapsulating
+ * a libcurl easy handle. It sets the target URL, HTTP method, optional headers,
+ * payload, and SSL/TLS credentials (CA certificate, client certificate, and private key).
+ * It also registers a write callback to store the server's response into a
+ * provided response stream buffer.
+ *
+ * Note: The returned object must be cleaned up by the caller using
+ * `gvm_http_free()` to free all associated resources. The request is not
+ * executed by this function — only configured.
+ *
+ * @param url           The full request URL.
+ * @param method        The HTTP method to use (GET, POST, etc.).
+ * @param payload       Optional request body for POST or PUT.
+ * @param headers       Optional custom headers (gvm_http_headers_t).
+ * @param ca_cert       Optional CA certificate for server verification.
+ * @param client_cert   Optional client certificate for mutual TLS.
+ * @param client_key    Optional client private key for mutual TLS.
+ * @param res           Response stream used as the write target during the request.
+ *
+ * @return A configured gvm_http_t object on success, or NULL on failure.
+ */
+gvm_http_t *
+gvm_http_init (const gchar *url, gvm_http_method_t method,
```

Comment:
I think this should be called `gvm_http_new`.

For me `_init` should be for a function that inits a structure that you give it.

Would be consistent with `gvm_http_multi_new`, `gvm_http_response_stream_new` and `gvm_http_headers_new`.